state_machine/applications/flight/Tasks/radio.py

```diff
@@ -9,24 +9,29 @@
 import radio_utils.headers as headers
 import radio_utils.message as message
 from pycubed import cubesat
+import settings
 import time
-from lib.alerts import alerts
+from alerts import alerts
 
 Message = message.Message
 
-ANTENNA_ATTACHED = False
-
 TX_SKIP = 5  # skip every TX_SKIP transmissions
 tx_ready_counter = 0
 
 TX_UPLINK_ENABLE_TIME = 5 * 60  # 5 minutes
 tx_before_time = 0
 
+def tx_allowed():
+    """
+    Return if we are allowed to transmit
+    """
+    return settings.TX_ALLOWED
```

Comment:
The `TX_ALLOWED` variable doesn't give us control over whether or not we listen for packets. It just controls if we can send packets. If the radio was initialized, but no antenna was attached there would be no point in listening for packets each time the radio task runs. Maybe, add an `RX_ALLOWED` variable? I like the ability to control RX and TX individually instead of just one `ANTENNA_ATTACHED` variable.

Reply:
Good idea.